service/matching/task_reader.go

```diff
@@ -149,6 +149,7 @@ dispatchLoop:
 				}
```

Comment:
comment on line 148 (right above this): this will log whenever the offer times out (1m). so for an invalid task with a recent create time, we may loop through here up to 10 times (and log 10 lines) before we verify and discard it? maybe we should include "context timed out" in the "too spammy" category (line 144)?

Reply:
added timeout / cancel error check, plus using throttled logger